src/checkInput.js

```diff
@@ -0,0 +1,99 @@
+import getMaxValue from 'ml-array-max';
+
+import { sumOfGaussianLorentzians } from './shapes/sumOfGaussianLorentzians';
+import { sumOfGaussians } from './shapes/sumOfGaussians';
+import { sumOfLorentzians } from './shapes/sumOfLorentzians';
+
+export function checkInput(data, options) {
+  let {
+    shape = { kind: 'gaussian' },
+    optimization = {
+      kind: 'lm',
+    },
+  } = options;
+
+  if (typeof shape.kind !== 'string') {
+    throw new Error('kind should be a string');
+  }
+
+  let kind = shape.kind.toLowerCase().replace(/[^a-z]/g, '');
+
+  let nbParams;
+  let paramsFunc;
+  switch (kind) {
+    case 'gaussian':
+      nbParams = 3;
```

Comment:
We should I think make distinction between defaultParameters and the actual parameters that we will optimize based on the options.
Allowed parameters could also define the default ratio. 
```
defaultParameters: {
  x: {minFactor: 0.5, maxFactor. 2, gradientDifferent: xxx},
  y: {minFactor: 0, maxFactor: 1, gradientDifferent: xxx}},
  width: {minFactor: 0.5, maxFactor: 1, gradientDifferent: xxx}},
}
```

Reply:
We make gradientDifference for x and width depending of peak width, so I think it should have not a default value. I prefer your format